### ISSUE TITLE
Search results page angular Uncaught Errors

### ISSUE THREAD

**Issue by racitup** [awesto/django-shop#469]
When I try to use the search function on my shop implementation I am getting the following two errors on the browser console:

`Uncaught SyntaxError: Unexpected token &`

```
Uncaught Error: [$injector:modulerr] Failed to instantiate module myShop due to:
Error: [$injector:nomod] Module 'myShop' is not available!
You either misspelled the module name or forgot to load it.
If registering a module ensure that you specify the dependencies as the second argument.
```

The first error appears to be caused by the `data|rest_json` in this line in the `results.html` template
`{% addtoblock "ng-config" %}['searchResultsProvider', function(provider) { provider.setSearchResults({{ data|rest_json }}); }]{% endaddtoblock %}`

The error seems to be caused by the first `&` in:
`provider.setSearchResults({&quot;count&quot;:1,&quot;next&quot;:null,&quot;previous&quot;:null,&quot;results&quot;:[{&quot;product_name&quot;:&quot;...`

Any ideas what might cause this?

**Comment by racitup** [comment]
It appears I need `{{ data|rest_json|safe }}` is this a bug or am I doing something wrong in my serializer?

**Comment by jrief** [comment]
The demo base template defines something like this:

```
	<script type="text/javascript">
	angular.module('myShop', ['ngAnimate', 'ngSanitize',
		{% render_block "ng-requires" postprocessor "djng.sekizai_processors.module_list" %}
	]).config(['$httpProvider', function($httpProvider) {
		$httpProvider.defaults.headers.common['X-CSRFToken'] = '{{ csrf_token }}';
		$httpProvider.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
	}]).config(['$locationProvider', function($locationProvider) {
		$locationProvider.html5Mode(false);
	}]){% render_block "ng-config" postprocessor "djng.sekizai_processors.module_config" %};
	</script>
```

I'm asking because I'm wondering about that error: ``Error: [$injector:nomod] Module 'myShop' is not available!``

**Comment by rfleschenberg** [comment]
@racitup  Good question. I also did `{{ data|rest_json|safe }}` in my project, but to be honest, I was not sure if this solution is correct, planned to look into it later and then forgot :)

@jrief: What do you think?

**Comment by racitup** [comment]
@rfleschenberg I don't really understand why an extra `|safe` fixes it because the `|rest_json` already returns with `return mark_safe(data)`

@jrief I think the 2nd Error is caused by the 1st Error. My base template is the same as you have quoted and was taken from the demo base template

**Comment by rfleschenberg** [comment]
I agree. Someone should analyze this further. I cannot promise to do it in the near future, unfortunately.

**Comment by racitup** [comment]
Having looked into this it appears to be because the `data = JSONRenderer().render(value)` returns a byte string rather than an actual string. So `return mark_safe(data)` in `rest_json` will return a `SafeBytes` object which seems to get escaped.
Using the `safe` filter first forces the input to text using `force_text` because of the `@stringfilter` decorator. So presumably ends up as a `SafeString` object which doesn't get escaped.

I've tried this out using `return mark_safe(force_text(data))` in `rest_json` and `{{ data|rest_json }}` and it works! Should I create a PR?

**Comment by jrief** [comment]
Thanks. Never encountered a bytestring here. Anyway I patched it and it will be part of 0.9.4.

**Comment by racitup** [comment]
Thanks Jacob!
